### PR TITLE
[SDPA-6182] Disable basic auth drupal shield on files.

### PR DESCRIPTION
**Pull request by edyuenyw**
**JIRA**
https://digital-engagement.atlassian.net/browse/SDPA-6182

**Change**
- Changes file path to exclude.

**Screenshots**
image

### CODE REVIEW

**Review by vincent-gao — APPROVED** [review]

**Review by vincent-gao — CHANGES_REQUESTED** [review]
I don't think it will work on the reference site, please double check @edyuenyw

**Comment by edyuenyw** [comment]
> I don't think it will work on the reference site, please double check @edyuenyw

@vincent-gao reference site doesn't seemed to have shield enable in the settings. Will discuss this further if required.

**Comment by edyuenyw** [comment]
> 

Additional info: Testing in reference site not available because shield is not enabled.

File: baywatch.install [hunk 1]
```diff
@@ -389,3 +389,16 @@ function baywatch_update_8018() {
...
   }
 }
+
+/**
+ * Corrects excluded files path in shield setting.
+ */
+function baywatch_update_8020() {
+  if (\Drupal::moduleHandler()->moduleExists('shield') === TRUE) {
+    $shield = \Drupal::configFactory()->getEditable('shield.settings');
+    $paths = $shield->get('paths');
```
Comment (MdNadimHossain) [inline baywatch.install:399]: please give a if condition to check if path exists then only do the str_replace

**Review by MdNadimHossain — CHANGES_REQUESTED** [review]
@edyuenyw two little changes. Look at the comment and the 2nd change is update the existing function to add the path in shield, so that when a new project get installed, it has the right path. Here is the place that needs to be updated - https://github.com/dpc-sdp/baywatch/blob/feature/SDPA-6182-disable-basic-auth-drupal-sheild-on-files-new/src/BaywatchOperation.php#L257

File: baywatch.install (hunk 1 above)
Comment (edyuenyw) [reply baywatch.install:399]: Will do.

**Comment by edyuenyw** [comment]
> 

Good to know. Will do.

**Review by MdNadimHossain — APPROVED** [review]

**Pull request merged**